src/lesson2-homework.ts

```diff
@@ -54,58 +66,101 @@ excercise6();
 // and returns a new array with the results of function called on each element of the array (function passed as a first parameter)
 function excercise7() {
   // TODO: add type annotations
-  function map(arr, fn) {
-    // TODO: add logic here
-    // TODO: use regular for loop
-    return [];
+  type Tfn = (a: number) => number;
+  function map(arr: number[], fn: Tfn) {
+    const result: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      result.push(fn(arr[i]));
+    }
+    return result;
   }
-  // TODO: create an array of numbers
-  // TODO: create a function which doubles a number
-  // TODO: call map function (created earlier) with the array and the function
-  // TODO: print the result to console
+
+  const arrOfNumbers = [10, 13, 8, 3, 11];
+  const doubleNumber = (num: number): number => {
+    return num * 2;
+  };
+  console.log(map(arrOfNumbers, doubleNumber));
 }
-// TODO: compile and run the code
+
 excercise7();
 
 // declare a function which takes a user and prits greeting to console
 function excercise8() {
-  // TODO: create a type for user, with name property
-  // TODO: create a function with name printGreeting, which takes a user and prits greeting to console
-  // TODO: create a type for product, with name property and price property
-  // TODO: create a product object, asign it some object literal
-  // TODO: call the function with product as a parameter
-  // TODO: call the function with object literal as a parameter
-  // TODO: try adding extra property to the object literal - observe the error
-  // TODO: fix the error with type assertion
+  type Tuser = { name: string };
+  const printGreeting = (user: Tuser): string => user.name;
+  type Tproduct = { name: string; price: number };
+  let product: Tproduct = {
+    name: 'Bread',
+    price: 11,
+  };
+  console.log(printGreeting(product));
+  console.log(printGreeting({ name: 'Bread', price: 11 }));
```

Comment:
// TODO: try adding extra property to the object literal - observe the error
  // TODO: fix the error with type assertion
  looks like this two requirements are not implemented

Reply:
// TODO: try adding extra property to the object literal - observe the error - this task was done on line 98.

Reply:
no, this is not object literal

Reply:
I understand object literal as something like this: { name: 'Bread', price: 11 }. Am I wrong?